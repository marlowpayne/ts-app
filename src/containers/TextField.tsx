import React from 'react';
import PropTypes from 'prop-types';

const noop = () => null;

export interface Props {
  onSubmit: Function;
}

interface State {
  text: string;
}

export class TextField extends React.Component<Props, State> {
  mount: boolean;

  propTypes = {
    onSubmit: PropTypes.func
  };

  defaultProps = {
    onSubmit: noop
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      text: ''
    };
  }

  componentDidMount() {
    this.mount = true;
  }

  componentWillUnmount() {
    this.mount = false;
  }

  onChange = (e: React.ChangeEvent<HTMLInputElement>): void => this.setState({ text: e.target.value });

  onSubmit = (e: React.MouseEvent<HTMLButtonElement>): void => {
    this.props.onSubmit(this.state.text);
    this.setState({ text: '' });
  }

  render() {
    return (
      <div>
        <input type="text" value={this.state.text} onChange={this.onChange} />
        <button type="submit" onClick={this.onSubmit}>
          Change Name
        </button>
      </div>
    );
  }
}
