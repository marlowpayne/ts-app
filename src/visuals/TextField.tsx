import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const noop = () => null;

export interface Props {
  onSubmit: Function;
}

interface State {
  text: string;
}

const Wrapper = styled.div``;

export class TextField extends React.Component<Props, State> {
  static propTypes = {
    onSubmit: PropTypes.func
  };

  static defaultProps = {
    onSubmit: noop
  };

  mount: boolean;

  constructor(props: Props) {
    super(props);
    this.state = {
      text: ''
    };
  }

  componentDidMount() {
    this.mount = true;
  }

  componentWillUnmount() {
    this.mount = false;
  }

  submitCurrentText = (): void => {
    this.props.onSubmit(this.state.text);
    this.setState({ text: '' });
  }

  onChange = (e: React.ChangeEvent<HTMLInputElement>): void => this.setState({ text: e.target.value });

  onSubmit = (e: React.MouseEvent<HTMLButtonElement>): void => this.submitCurrentText();

  onKeyUp = (e: React.KeyboardEvent<HTMLInputElement>): void => {
    if (e.key === 'Enter') {
      this.submitCurrentText();
    }
  }

  render() {
    return (
      <Wrapper>
        <input type="text" value={this.state.text} onChange={this.onChange} onKeyUp={this.onKeyUp} />
        <button type="submit" onClick={this.onSubmit}>
          Change Name
        </button>
      </Wrapper>
    );
  }
}
