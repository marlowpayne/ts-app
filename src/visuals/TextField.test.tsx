import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { TextField, Props } from './TextField';

const testProps: Props = {
  onSubmit: () => null,
};

describe('TextField visual comp', () => {
  it('shallow renders without crashing', () => {
    const el = shallow(<TextField {...testProps} />);
    expect(el.length).toEqual(1);
  });
});

describe('TextField visual comp -- Snapshot', () => {
  it('captures a snapshot', () => {
    const rendered = renderer.create(<TextField {...testProps} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
