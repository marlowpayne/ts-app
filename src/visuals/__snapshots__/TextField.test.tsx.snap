// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField visual comp -- Snapshot captures a snapshot 1`] = `
<div
  className="sc-bdVaJa bDWFJH"
>
  <input
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Change Name
  </button>
</div>
`;
